Core/Src/thermistor.c

```diff
@@ -46,22 +48,36 @@ double getTemperature(double voltageReading){		// USING STEINHART & HART EQUATIO
 }
 
 void StartReadTempTask(void const * argument){
-//	char msg[512];
 	char tempMsg[50];
+	char* time;
 
 	for (;;){
-		HAL_ADC_Start_DMA(&hadc1, ADC_Readings, 16);
-		while(!newData_thermistor) {}
-		newData_thermistor = 0;
+		if (newData_thermistor == 1) {
+			for(int i = 0; i < 16; i++) {
+				  temperatures[i] = getTemperature(ADC_TO_Voltage * ADC_Readings[i]);
+			}
 
-		for(int i = 0; i < 16; i++) {
-			  temperatures[i] = getTemperature(ADC_TO_Voltage * ADC_Readings[i]);
+			/* TODO SCU#35 */
+			/* Logging Starts */
+			time = get_time();
+			HAL_USART_Transmit(&husart1, (uint8_t *) time, strlen(time), 10);
+
+			sprintf(tempMsg, ",,%f,", temperatures[0]);
+			HAL_USART_Transmit(&husart1, (uint8_t *) tempMsg, strlen(tempMsg), 10);
+
+			sprintf(tempMsg, "%f,", temperatures[1]);
+			HAL_USART_Transmit(&husart1, (uint8_t *) tempMsg, strlen(tempMsg), 10);
+
+			sprintf(tempMsg, "%f,", temperatures[2]);
+			HAL_USART_Transmit(&husart1, (uint8_t *) tempMsg, strlen(tempMsg), 10);
+
+			sprintf(tempMsg, "%f\r\n", temperatures[3]);
+			HAL_USART_Transmit(&husart1, (uint8_t *) tempMsg, strlen(tempMsg), 10);
+			/* Logging Ends */
+
+			newData_thermistor = 0;
```

Comment:
Should probably take this out of the for loop as you only need to reset the flag once at the end.